### PR TITLE
feat(linter): implement `useDateNow`

### PR DESCRIPTION
**Pull request by minht11**
## Summary

Closes https://github.com/biomejs/biome/issues/2917
Implement [unicorn/prefer-date-now](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-date-now.md)

## Test Plan

Added snapshots

### CODE REVIEW

File: crates/biome_js_analyze/src/lint/nursery/use_date_now.rs [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
+        if does_node_needs_space_before_child(&node.syntax().parent()?) {
+            // Make fake token to get leading trivia
+            let leading_trivia = make::token_decorated_with_space(T![=])
+                .leading_trivia()
+                .pieces();
+
+            // In case `await +a` this produces double  space `await  +a`.
+            // TODO. Find a way to avoid this.
+            date_now_expr = date_now_expr
```
Comment (minht11) [inline crates/biome_js_analyze/src/lint/nursery/use_date_now.rs:122]: I wish there was more ergnomic way to fix this. I imagine a lot of rules are/will be impacted by this. Unicorn provides [fixSpaceAroundKeyword](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/810575620e3d9d90c60c2d6c5f2098c0a7b0348d/rules/fix/fix-space-around-keywords.js#L12-L34) but not sure how would that fit into biome.

**Review by ematipico — APPROVED** [review]

File: crates/biome_js_analyze/src/lint/nursery/use_date_now.rs (hunk 1 above)
Comment (ematipico) [reply crates/biome_js_analyze/src/lint/nursery/use_date_now.rs:122]: The formatter, that's for sure

**Review by ematipico — APPROVED** [review]

File: crates/biome_js_analyze/src/lint/nursery/use_date_now.rs (hunk 1 above)
Comment (minht11) [reply crates/biome_js_analyze/src/lint/nursery/use_date_now.rs:122]: You mean we would call formatter from lint rules? Conceptually it makes sense. Is there any precedent currently in the codebase?

Comment (ematipico) [reply crates/biome_js_analyze/src/lint/nursery/use_date_now.rs:122]: I might have misunderstood the comment, I'm sorry about that. You can ignore my comment

**Comment by ematipico** [comment]
Shoot we forgot to merge it and now there are conflicts 😓

**Comment by minht11** [comment]
No worries. Fixed conflicts 😁

**Pull request merged**